src/wallet/wallet2.cpp

```diff
@@ -889,14 +929,14 @@ void wallet2::expand_subaddresses(const cryptonote::subaddress_index& index)
     cryptonote::subaddress_index index2;
     for (index2.major = m_subaddress_labels.size(); index2.major < index.major + m_subaddress_lookahead_major; ++index2.major)
     {
-      for (index2.minor = 0; index2.minor < (index2.major == index.major ? index.minor : 0) + m_subaddress_lookahead_minor; ++index2.minor)
+      uint32_t end = (index2.major == index.major ? index.minor : 0) + m_subaddress_lookahead_minor;
+      index2.minor = 0;
+      std::vector<crypto::public_key> pkeys = get_subaddress_spend_public_keys(index2.major, index2.minor, end);
+      for (uint32_t begin = index2.minor; index2.minor < end; ++index2.minor)
```

Comment:
I find this part a little odd because `begin` is always 0 and not really used. How about this for the sake of clarity?
```c++
      const uint32_t begin = 0;
      const std::vector<crypto::public_key> pkeys = get_subaddress_spend_public_keys(index2.major, begin, end);
      for (index2.minor = begin; index2.minor < end; ++index2.minor)
``` 
Or we could just replace `begin` with 0.

Also, `const` could be added when appropriate.

Reply:
Well, begin was not always 0 originally, since it was testing for already calculated data first. It can be changed now, since it always recalculates.